library/std/src/os/horizon/raw.rs

```diff
@@ -0,0 +1,70 @@
+//! Horizon OS raw type definitions
+
+#![stable(feature = "raw_ext", since = "1.1.0")]
+#![deprecated(
```

Comment:
Should avoid adding insta-deprecated code. This is a green-field target so it is not constrained by any stability guarantees. Just make these definitions private if these are used in the implementation, or don't add them at all.

Reply:
(For `unix::raw` you can just `cfg` the re-exports out)

Reply:
Sure. I was following ESP-IDF: https://github.com/rust-lang/rust/pull/87666

Reply:
Updated so the `raw` module is `pub(crate)` (some of the definitions are still used by std).

Reply:
With the module marked as `pub(crate)` the stability annotations should no longer be necessary.

It would be also ideal if the items in this module were marked as explicitly `pub(crate)` to ensure they do not appear in the public API elsewhere.

Reply:
I tried changing them to `pub(crate)`, but most of them are used elsewhere:
```
error[E0365]: `pthread_t` is only public within the crate, and cannot be re-exported outside
  --> library/std/src/os/unix/raw.rs:27:9
   |
27 | pub use super::platform::raw::pthread_t;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ re-export of crate public `pthread_t`
   |
   = note: consider declaring type or module `pthread_t` with `pub`
```

Just removing the stability attributes but keeping the `pub` visibility caused other errors:
```
error: type alias has missing stability attribute
  --> library/std/src/os/horizon/raw.rs:10:1
   |
10 | pub type pthread_t = libc::pthread_t;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

So I don't think I can change this.

Reply:
So, my thinking was that you might have wanted to `#[cfg(not(...)]` these deprecated re-exports, but having thought about it more the way it is right now is also okay – consistency between platforms has some value and it isn't too onerous to provide it.

Reply:
Ok, I'll keep the os raw module as `pub(crate)` since we can always change that later.